crates/pera-core/src/dwallet_mpc/mpc_session.rs

```diff
@@ -182,31 +289,39 @@ impl DWalletMPCSession {
     }
 
     /// Stores a message in the pending messages map.
-    /// The code stores every new message it receives for that session,
-    /// and when we reach the end of delivery,
-    /// we will advance the session if we have a threshold of messages.
-    fn store_message(&mut self, round: usize, message: &DWalletMPCMessage) -> DwalletMPCResult<()> {
-        let epoch_store = self.epoch_store()?;
-        let party_id = authority_name_to_party_id(&message.authority, &epoch_store)?;
-        if self.pending_messages[round].contains_key(&party_id) {
-            return Err(DwalletMPCError::MaliciousParties(vec![party_id]));
+    /// Every new message received for a session is stored.
+    /// When a threshold of messages is reached, the session advances.
+    fn store_message(&mut self, message: &DWalletMPCMessage) -> DwalletMPCResult<()> {
+        let source_party_id =
+            authority_name_to_party_id(&message.authority, &*self.epoch_store()?)?;
+
+        let msg_len = self.pending_messages.len();
```

Comment:
```suggestion
        let current_round = self.pending_messages.len();
```

Reply:
This is definitely a wrong name: `msg_len` means the length of a message, not of a vector of messages.